src/store/annotations/reducers.ts

```diff
@@ -1,6 +1,23 @@
+import {CREATE_ANNOTATION} from './actions';
 
-const initialState = [];
+const initialState = {
+  data: [],
+};
+
+function maxId(state) {
+  return Math.max(0, ...state.data.map(annotation => annotation.annotationId)) + 1;
+}
 
 export default function editor(state = initialState, action) {
-  return state;
+  switch (action.type) {
+    case CREATE_ANNOTATION:
+      return Object.assign({}, {
```

Comment:
I'd propose the following, a bit simpler and future-proof (won't need to modify it if any new fields gets added to this reducer)
```
return {
  ...state,
  data: state.data.concat({
    ...action.payload,
    annotationId: maxId(state),
  });
};
```

Reply:
and maybe move it to `createAnnotationActionHandler`, similarly as in `textSelector/reducers.ts` ?